### PR TITLE
Changes to epoch-related members

### PR DESCRIPTION
**Pull request by jskeet**
This fixes issue #371.

// cc @mj1856

### CODE REVIEW

File: src/NodaTime.Test/InstantTest.cs [hunk 1]
```diff
@@ -77,48 +77,76 @@ public void WithOffset_NonIsoCalendar()
...
+            Assert.Throws<ArgumentOutOfRangeException>(() => Instant.FromUnixTimeSeconds(long.MinValue / 1000000));
+        }
+
+        [Test]
+        [TestCase(-1500, -2)]
+        [TestCase(-1000, -1)]
+        [TestCase(-500, -1)]
+        [TestCase(0, 0)]
+        [TestCase(500, 0)]
+        [TestCase(1000, 1)]
```
Comment (malcolmr) [inline src/NodaTime.Test/InstantTest.cs]: worth having e.g. 999, 1001 here (perhaps negative too?)

File: src/NodaTime/Instant.cs [hunk 2]
```diff
@@ -570,12 +558,12 @@ public static Instant FromSecondsSinceUnixEpoch(long seconds)
...
         /// <returns>An <see cref="Instant"/> at exactly the given number of milliseconds since the Unix epoch.</returns>
         /// <exception cref="ArgumentOutOfRangeException">The constructed instant would be out of the range representable in Noda Time.</exception>
-        public static Instant FromMillisecondsSinceUnixEpoch(long milliseconds)
+        public static Instant FromUnixTimeMilliseconds(long milliseconds)
         {
             Preconditions.CheckArgumentRange(nameof(milliseconds), milliseconds, MinMilliseconds, MaxMilliseconds);
             return Instant.FromTrustedDuration(Duration.FromMilliseconds(milliseconds));
         }
-
+
```
Comment (malcolmr) [inline src/NodaTime/Instant.cs]: strange extra spaces here.

File: src/NodaTime/Instant.cs [hunk 3]
```diff
@@ -584,13 +572,52 @@ public static Instant FromMillisecondsSinceUnixEpoch(long milliseconds)
         /// intent more explicitly.</remarks>
         /// <returns>An <see cref="Instant"/> at exactly the given number of ticks since the Unix epoch.</returns>
         /// <param name="ticks">Number of ticks since the Unix epoch. May be negative (for instants before the epoch).</param>
-        public static Instant FromTicksSinceUnixEpoch(long ticks)
+        [Pure]
```
Comment (malcolmr) [inline src/NodaTime/Instant.cs]: static [Pure] methods don't make sense, do they?

File: www/unstable/userguide/migration-to-2.md [hunk 4]
```diff
@@ -72,6 +72,15 @@ Factory methods in `CalendarSystem` which either didn't take any parameters (`Ge
 no longer support those parameters (`GetCopticCalendar`, `GetJulianCalendar`) have been converted into properties.
 So for example, the equivalent of `CalendarSystem.GetJulianCalendar(4)` is now just `CalendarSystem.Julian`.
 
+Methods and properties on `Instant` to do with the Unix epoch have been renamed to be consistent with
+`DateTimeOffset`:
```
Comment (malcolmr) [inline www/unstable/userguide/migration-to-2.md]: Perhaps "consistent with new `DateTimeOffset` methods added in ..."

File: www/unstable/userguide/migration-to-2.md [hunk 5]
```diff
@@ -72,6 +72,15 @@ Factory methods in `CalendarSystem` which either didn't take any parameters (`Ge
...
+
+- `FromSecondsSinceUnixEpoch` is now `FromUnixTimeSeconds`
```
Comment (malcolmr) [inline www/unstable/userguide/migration-to-2.md]: perhaps `FromSecondsSinceUnixEpoch()` throughout this para.

File: www/unstable/userguide/versions.md [hunk 6]
```diff
@@ -47,6 +47,10 @@ See the [Noda Time 1.x to 2.0 migration guide](migration-to-2.html) for full det
 - `y` and `yyy` are no longer supported in date format specifiers; use `yy` or `yyyy` instead.
 - The date format specifiers `yy` and `yyyy` now refer to the year of era instead of the absolute year; `u` is used for absolute year.
 - Some parameters have been renamed for consistency, affecting code which uses named arguments.
+- Factory methods for converting from units since the Unix epoch to `Instant` have been renamed to `FromUnixTimeXyz`
```
Comment (malcolmr) [inline www/unstable/userguide/versions.md]: Likewise refer to methods as FooMethod()?

**Comment by malcolmr** [comment]
lgtm

**Comment by mj1856** [comment]
:+1:

File: src/NodaTime/Instant.cs (hunk 2 above)
Comment (jskeet) [inline src/NodaTime/Instant.cs]: Odd. Done.

File: src/NodaTime.Test/InstantTest.cs (hunk 1 above)
Comment (jskeet) [inline src/NodaTime.Test/InstantTest.cs]: Done.

File: src/NodaTime/Instant.cs (hunk 3 above)
Comment (jskeet) [inline src/NodaTime/Instant.cs]: Nope. No idea what I was thinking.

File: www/unstable/userguide/migration-to-2.md (hunk 4 above)
Comment (jskeet) [inline www/unstable/userguide/migration-to-2.md]: Done.

File: www/unstable/userguide/migration-to-2.md (hunk 5 above)
Comment (jskeet) [inline www/unstable/userguide/migration-to-2.md]: Done

File: www/unstable/userguide/versions.md (hunk 6 above)
Comment (jskeet) [inline www/unstable/userguide/versions.md]: Done

**Comment by jskeet** [comment]
Have made the changes in a separate commit; would squash, but distributing them amongst the original three commits would be tricky.

File: src/NodaTime/Instant.cs [hunk 7]
```diff
@@ -584,13 +572,51 @@ public static Instant FromMillisecondsSinceUnixEpoch(long milliseconds)
...
+
+        /// <summary>
+        /// Gets the number of milliseconds since the Unix epoch. Negative values represent instants before the Unix epoch.
+        /// </summary>
+        /// <remarks>
+        /// If the number of nanoseconds in this instant is not an exact number of milliseconds, the value is truncated towards the start of time.
+        /// </remarks>
+        /// <value>The number of milliseconds since the Unix epoch.</value>
+        [Pure]
+        public long ToUnixTimeMilliseconds()
```
Comment (malcolmr) [inline src/NodaTime/Instant.cs]: Why not => here and above?

**Comment by malcolmr** [comment]
all LGTM otherwise.

File: src/NodaTime/Instant.cs (hunk 7 above)
Comment (jskeet) [inline src/NodaTime/Instant.cs]: No reason at all - thanks :)

**Pull request merged**